### PR TITLE
JENA-1398: Marked archaic FOAF terms in the vocabulary as deprecated.

### PR DESCRIPTION
**Pull request by shirren**
@afs The approach was to download the latest FOAF.rdf from the W3C website from which the FOAF.ttl and FOAF.java file was generated. "Archaic" terms from the spec were then marked as @Deprecated in the Java code. All tests pass. Please provide feedback or comments, thanks in advance.

### CODE REVIEW

File: jena-arq/src/main/java/org/apache/jena/sparql/vocabulary/FOAF.java [hunk 1]
```diff
@@ -19,242 +19,282 @@
...
+
+    /** <p>A checksum for the DNA of some thing. Joke.</p> */
+    @Deprecated
+    public static final Property dnaChecksum = M_MODEL.createProperty( "http://xmlns.com/foaf/0.1/dnaChecksum" );
+
+    /** <p>The family name of some person.</p> */
+    public static final Property familyName = M_MODEL.createProperty( "http://xmlns.com/foaf/0.1/familyName" );
+
+    /** <p>The family name of some person.</p> */
+    @Deprecated
```
Comment (rvesse) [inline jena-arq/src/main/java/org/apache/jena/sparql/vocabulary/FOAF.java]: For deprecated terms like this where there is a new alternative can we explicitly point to the new term to help users migrate forwards

Comment (shirren) [reply jena-arq/src/main/java/org/apache/jena/sparql/vocabulary/FOAF.java]: @rvesse that's a good idea. I'll make it so.

**Comment by shirren** [comment]
@rvesse I've made those changes. How does the Jena team schedule releases, and do you know when this PR will be released? Thank you so much for working on this projects, it's been invaluable to our team.

**Comment by rvesse** [comment]
Thanks for the updates, looks good to me

We aim to release approximately every three months, the last release was July so we are probably due another release soon. However everything is done on volunteer time so it will really depend on peoples availability

**Comment by shirren** [comment]
@rvesse understand, thanks once again.

**Pull request merged**